packages/wallets/src/components/Base/WalletButton/WalletButton.tsx

```diff
@@ -1,35 +1,76 @@
-import React, { ComponentProps, CSSProperties } from 'react';
+import React, { ButtonHTMLAttributes } from 'react';
 import classNames from 'classnames';
+import { TGenericSizes } from '../types';
+import { WalletText } from '../WalletText';
 import styles from './WalletButton.module.css';
 
-interface WalletButtonProps extends ComponentProps<'button'> {
-    children: React.ReactNode;
-    color?: CSSProperties['color'] | 'primary-light' | 'primary';
+interface WalletButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+    color?: 'black' | 'primary-light' | 'primary' | 'white';
+    icon?: React.ReactNode;
     isRounded?: boolean;
-    size?: 'lg' | 'md' | 'sm';
-    variant?: 'contained' | 'outlined' | 'text';
+    size?: Omit<TGenericSizes, 'lg' | 'md' | 'sm'>;
```

Comment:
```suggestion
    size?: Pick<TGenericSizes, 'lg' | 'md' | 'sm'>;
```
Shouldn't this be Pick?